diesel/src/query_builder/select_statement/dsl_impls.rs

```diff
@@ -470,3 +469,51 @@ where
         InsertFromSelect::new(self)
     }
 }
+
+impl<'a, F, S, D, W, O, L, Of, G> NullableSelectDsl
+    for SelectStatement<F, SelectClause<S>, D, W, O, L, Of, G>
+where
+    SelectClause<S>: SelectClauseExpression<F>,
+    <SelectClause<S> as SelectClauseExpression<F>>::SelectClauseSqlType: NotNull,
```

Comment:
I'm not clear why this is needed. We allow `.nullable` on expressions which are already nullable, and `Nullable<T>` works fine regardless of the SQL type for `T`. I'd rather be consistent and not have this bound.